sklego/preprocessing.py

```diff
@@ -206,3 +206,55 @@ def _check_X_for_type(X):
         """Checks if input of the Selector is of the required dtype"""
         if not isinstance(X, pd.DataFrame):
             raise TypeError("Provided variable X is not of type pandas.DataFrame")
+
+
+class OrthogonalTransformer(BaseEstimator, TransformerMixin):
+    """
+    Transform the columns of a dataframe or numpy array to a column orthogonal or orthonormal matrix.
+    Q, R such that X = Q*R, with Q orthogonal, from which follows Q = X*inv(R)
+    :param normalize: whether orthogonal matrix should be orthonormal as well
+    """
+
+    def __init__(self, normalize=False):
+        self.normalize = normalize
+
+    def fit(self, X, y=None):
+        """
+        Store the inverse of R of the QR decomposition of X, which can be used to calculate the orthogonal projection
+        of X. If normalization is required, also stores a vector with normalization terms
+        """
+        input_is_df = isinstance(X, pd.DataFrame)
```

Comment:
sklearn has a `check_array` function that will convert the dataframe for you and also performs some checks. Also, do we have to convert to a numpy array? I kinda expect all the code you've written to work agnostically of whether it's a dataframe or an array anyway.

Reply:
In the latest commit, I have added the check_array for converting to np array. Do we want to return a `pd.DataFrame` if that is the supplied type of `X`?

Reply:
That would indeed be preferred. The easiest would probably be to just call `check_array` but not reassign it to X

Reply:
Matrix multiplication always results in a numpy array, so if the goal is that a df is returned if a df is the input, then conversion is necessary. (In `fit`, matrixmultiplication is always done). 

Do you want me to add this conversion?

Reply:
`fit` shouldn't change the input X anyway right?